packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/search_dsl/query_params.test.ts

```diff
@@ -235,6 +236,38 @@ describe('#getQueryParams', () => {
         expect(filters.some((filter) => filter.references_filter === true)).toBeDefined();
       });
     });
+
+    describe('`hasNoReference` parameter', () => {
```

Comment:
(honest question) Would it make sense to have a test asserting behavior when both `hasReference` and `hasNoReference` are specified?

Reply:
Good point. 👍  I've added 2 tests in https://github.com/elastic/kibana/pull/142479/commits/a4da3f32119cdf3e91c85ef75927dbb0581f8c16 to make sure we have the expected behaviour when both `has_reference` and `has_no_reference` are provided.